personalization.admin.inc

```diff
@@ -125,21 +125,58 @@ function personalization_admin() {
     '#title' => t('Geographical based personalization'),
   );
 
-  $form['geographic_based']['pz_geo'] = array(
-    '#type' => 'checkboxes',
-    '#title' => t("How should the user's location be ascertained"),
-    '#default_value' => $settings['pz_geo'],
-    '#options' => array(
-      'from_api' => t('Request location from the users IP automatically from http://www.telize.com/ API'),
-      'ask_user' => t('Ask the user for their location.'),
+  $form['geographic_based']['pz_geo_enabled'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Enable loading geo location data'),
+    '#default_value' => !empty($settings['pz_geo_enabled']) ? $settings['pz_geo_enabled'] : 0,
+  );
+
+  // Define states for showing/hiding geo location settings.
+  $pz_geo_enabled_state = array(
+    'visible' => array(
+      ':input[name="pz_geo_enabled"]' => array(
+        'checked' => TRUE,
+      )
     ),
-    '#description' => t("If you select both options the user will only be asked if http://www.telize.com/ cannot ascertain the location. Requesting the location from the user will require them to be using a HTML5 compliant browser and your Drupal theme to be HTML5."),
   );
 
-  // Settings visibility conditions for showing geo related weights.
-  $geo_state_one_of_selected = array(
-    array(':input[name="pz_geo[from_api]"]' => array('checked' => TRUE)),
-    array(':input[name="pz_geo[ask_user]"]' => array('checked' => TRUE)),
+  // Load options depending on installed modules.
+  $pz_geo_options = array(
+    'default' => t('Default'),
+  );
+
+  // @TODO: rebuild loading options with hook for extending.
+  // Check if Smart IP is installed.
+  if (module_exists('smart_ip')) {
+    // Don't translate title of the module.
+    $pz_geo_options['smart_ip'] = 'Smart IP';
+  }
+
+  // Check if Smart IP is installed.
+  if (module_exists('ip_geoloc')) {
+    // Don't translate title of the module.
+    $pz_geo_options['ip_geoloc'] = 'IP Geolocation Views & Maps';
+  }
+
+  $pz_geo_description = t("
+<ul>
+  <li><strong>Default</strong> - requesting the location from the user will require them to be using a HTML5 compliant browser and your Drupal theme to be HTML5</li>
+  <li><strong>Smart IP</strong> - geo location data are loaded from !smart_ip_link module.</li>
+  <li><strong>IP Geolocation Views & Maps</strong> - geo location data are loaded from !ip_geoloc_link module.</li>
+</ul>
+If you don't see the option make sure corresponding module is installed. Please check README.txt for details.
+", array(
+    '!smart_ip_link' => l('Smart IP', 'https://www.drupal.org/project/smart_ip', array('attributes' => array('target' => '_blank'))),
+    '!ip_geoloc_link' => l('IP Geolocation Views & Maps', 'https://www.drupal.org/project/ip_geoloc', array('attributes' => array('target' => '_blank'))),
+  ));
```

Comment:
Would be nicer as a them('item_list' but not a blocker

Reply:
Done.